framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1146,6 +1152,41 @@ public AnnotatedTypeMirror getAnnotatedTypeFromTypeTree(Tree tree) {
         return type;
     }
 
+    /** Returns the set of qualifiers that are the upper bounds for a use of the type. */
+    public Set<AnnotationMirror> getTypeDeclarationBounds(AnnotatedTypeMirror type) {
+        return qualifierUpperBounds.getBoundQualifiers(type);
+    }
+
+    /**
+     * Returns the set of qualifiers that are the upper bound for a type use if no other bound is
+     * specified for the type.
+     *
+     * <p>Super implementation returns the top qualifiers by default. Subclass may override to
```

Comment:
The `Super` made me look for a super class implementation. Wouldn't `This implementation...` work better?